pkg/deploy/controller/imagechange/controller.go

```diff
@@ -120,17 +134,18 @@ func (c *ImageChangeController) Handle(stream *imageapi.ImageStream) error {
 	// Attempt to regenerate all configs which may contain image updates
 	anyFailed := false
 	for _, config := range configsToUpdate {
-		if _, err := c.client.DeploymentConfigs(config.Namespace).Update(config); err != nil {
+		if updated, err := c.dn.DeploymentConfigs(config.Namespace).Update(config); err != nil {
```

Comment:
Wow, the shenanigans involved in not mutated the cache above here in the method just about blew my mind.  Can I at least get an assignment to `configs[i]` above instead of `config =`?